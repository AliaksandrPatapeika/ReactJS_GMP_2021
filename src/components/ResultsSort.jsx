import '../less/components/ResultsSort.less';

import React, {useState} from 'react';

const ResultsSort = () => {
  const [sortOrderAsc, setSortOrder] = useState(false);
  const className = sortOrderAsc ? 'arrow arrow-up' : 'arrow arrow-down';
  const toggleSortSortOrder = () => {
    setSortOrder(!sortOrderAsc);
  };

  return (
    <div className="resultsSortContainer">
      <div className="sortBy">SORT BY</div>
      <div className="releaseDate">RELEASE DATE</div>
      <div
        role="button"
        tabIndex="0"
        aria-label="Sort Button"
        className={className}
        onClick={toggleSortSortOrder}
        onKeyDown={toggleSortSortOrder}
      />
    </div>
  );
};

export default ResultsSort;
