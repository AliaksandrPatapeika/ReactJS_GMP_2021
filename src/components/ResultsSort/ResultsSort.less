@import "../../less/variables";

.resultsSortContainer {
	display: flex;

	.sortBy {
		color: @gray;
		font-size: 20px;
		font-family: @NetflixSans-Regular;
	}

	.releaseDate {
		margin-left: 30px;
		color: @white;
		font-size: 36px;
		font-family: @BloomingElegant_Sans;
		line-height: 0.23;
	}

	.arrow {
		cursor: pointer;
		width: 14px;
		height: 14px;
		margin-left: 16px;
		background-color: transparent;
		margin-top: 8px;

		&:hover {
			filter: contrast(190%);
		}
	}

	.arrow-up {
		transform: rotate(180deg);
	}
}