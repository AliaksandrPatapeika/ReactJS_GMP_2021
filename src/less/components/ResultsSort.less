@import "../variables";

.resultsSortContainer {
	display: flex;

	.sortBy {
		font-size: 20px;
		color: @gray;
		font-family: @NetflixSans-Regular;
	}

	.releaseDate {
		color: @white;
		font-family: @BloomingElegant_Sans;
		font-size: 35px;
		line-height: 0.23;
		margin-left: 30px;
	}

	.arrow {
		border-left: 7px solid transparent;
		border-right: 7px solid transparent;
		cursor: pointer;
		outline: none;
		margin-left: 16px;
	}

	.arrow-up {
		margin-bottom: 28px;
		border-bottom: 8px solid @red;
	}

	.arrow-down {
		margin-top: 12px;
		border-top: 8px solid @red;
	}
}



