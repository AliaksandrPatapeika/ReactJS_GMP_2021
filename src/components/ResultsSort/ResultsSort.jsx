import './ResultsSort.less';

import React, {useState} from 'react';

import arrowDown from '../../img/arrowDown.svg';
import Button from '../Button';

const ResultsSort = () => {
  const [sortOrderAsc, setSortOrder] = useState(false);
  const arrowClassName = sortOrderAsc ? 'arrow arrow-up' : 'arrow';
  const toggleSortSortOrder = () => {
    setSortOrder(!sortOrderAsc);
  };

  return (
    <div className="resultsSortContainer">
      <div className="sortBy">SORT BY</div>
      <div className="releaseDate">RELEASE DATE</div>
      <Button className={arrowClassName} onClick={toggleSortSortOrder}>
        <img src={arrowDown} alt="Sort Button" />
      </Button>
    </div>
  );
};

export default ResultsSort;
