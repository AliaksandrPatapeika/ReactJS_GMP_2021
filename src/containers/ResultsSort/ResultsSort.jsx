import './ResultsSort.less';

import {head} from 'lodash';
import React, {useState} from 'react';

import Button from '../../components/Button';
import Select from '../../components/Select';
import SortIcon from '../../components/SortIcon';
import {SortBy} from '../../constants';
import useToggle from '../../hooks';

const ResultsSort = () => {
  const [sortBy, setSortBy] = useState(head(SortBy));
  const [sortOrderAsc, toggleSortOrder] = useToggle(false);

  const setSortByHandler = (selectItem) => {
    setSortBy(selectItem);

    // TODO with Redux dispatch(sortByValue(selectItem.value));
  };

  return (
    <div className="resultsSortContainer">
      <div className="sortBy">SORT BY</div>
      <Select
        className="select resultSortSelect"
        classNamePrefix="select"
        id="resultSort"
        options={SortBy}
        onChange={setSortByHandler}
        value={sortBy}
      />
      <Button className="arrow" onClick={toggleSortOrder}>
        <SortIcon sortOrderAsc={sortOrderAsc} />
      </Button>
    </div>
  );
};

export default ResultsSort;
