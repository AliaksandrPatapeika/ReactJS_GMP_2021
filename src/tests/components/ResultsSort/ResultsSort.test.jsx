import React from 'react';
import renderer from 'react-test-renderer';

import ResultsSort from '../../../components/ResultsSort';

describe('ResultsSort', () => {
  test('renders correctly', () => {
    const resultsSort = renderer
      .create(<ResultsSort />)
      .toJSON();

    expect(resultsSort).toMatchSnapshot();
  });
});
