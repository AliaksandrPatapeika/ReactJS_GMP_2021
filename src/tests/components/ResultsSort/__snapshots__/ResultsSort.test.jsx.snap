// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsSort renders correctly 1`] = `
<div
  className="resultsSortContainer"
>
  <div
    className="sortBy"
  >
    SORT BY
  </div>
  <div
    className="releaseDate"
  >
    RELEASE DATE
  </div>
  <div
    aria-label="Sort Button"
    className="arrow arrow-down"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  />
</div>
`;
